ffi/capi/src/errors.rs

```diff
@@ -11,20 +12,91 @@ pub mod ffi {
 
     #[derive(Debug, PartialEq, Eq)]
     #[repr(C)]
-    /// A common enum for errors that ICU4X may return, organized by API
-    ///
-    /// The error names are stable and can be checked against as strings in the JS API
-    #[diplomat::rust_link(fixed_decimal::ParseError, Enum, compact)]
+    #[diplomat::rust_link(icu::provider::DataError, Struct, compact)]
+    #[diplomat::rust_link(icu::provider::DataErrorKind, Enum, compact)]
+    pub enum ICU4XDataError {
+        Unknown = 0x00,
+        MissingDataMarker = 0x01,
+        MissingVariant = 0x02,
+        MissingLocale = 0x03,
+        NeedsVariant = 0x04,
+        NeedsLocale = 0x05,
+        ExtraneousLocale = 0x06,
+        FilteredResource = 0x07,
+        MismatchedType = 0x08,
+        MissingPayload = 0x09,
+        InvalidState = 0x0A,
+        Custom = 0x0B,
+        Io = 0x0C,
+        UnavailableBufferFormat = 0x0D,
+        MismatchedAnyBuffer = 0x0E,
+        DataStructValidityError = 0x0F,
+    }
+
+    #[derive(Debug, PartialEq, Eq)]
+    #[repr(C)]
+    #[diplomat::rust_link(icu::locale::ParseError, Enum, compact)]
+    pub enum ICU4XLocaleParseError {
+        Unknown = 0x00,
+        Language = 0x01,
+        Subtag = 0x02,
+        Extension = 0x03,
+    }
+
+    #[derive(Debug, PartialEq, Eq)]
+    #[repr(C)]
+    #[diplomat::rust_link(fixed_decimal::ParseError, Struct, compact)]
+    pub enum ICU4XFixedDecimalParseError {
+        Unknown = 0x00,
+        Limit = 0x01,
+        Syntax = 0x02,
+    }
+
+    #[derive(Debug, PartialEq, Eq)]
+    #[repr(C)]
     #[diplomat::rust_link(fixed_decimal::LimitError, Struct, compact)]
+    pub enum ICU4XFixedDecimalLimitError {
+        TodoZst,
```

Comment:
question: do we have plans for this, or is this the permanent name?

If these errors are permanently ZSTs I think we should just return `()`, yes?

Reply:
Okay

Reply:
https://github.com/rust-diplomat/diplomat/pull/494 adds support, using it is blocked on JS2